src/msw/uilocale.cpp

```diff
@@ -140,11 +272,86 @@ class wxUILocaleImplLCID : public wxUILocaleImpl
         return buf;
     }
 
+    wxLocaleIdent GetLocaleId() const wxOVERRIDE
+    {
+        return wxLocaleIdent::FromTag(GetName());
+    }
+
     wxString GetInfo(wxLocaleInfo index, wxLocaleCategory cat) const wxOVERRIDE
     {
         return wxGetInfoFromLCID(m_lcid, index, cat);
     }
 
+    wxString GetLocalizedName(wxLocaleName name, wxLocaleForm form) const wxOVERRIDE
+    {
+        wxChar buf[256];
+        buf[0] = wxT('\0');
+        wxString str;
+        switch (name)
+        {
+            case wxLOCALE_NAME_LOCALE:
+                switch (form)
+                {
+                    case wxLOCALE_FORM_NATIVE:
+                        {
+                            ::GetLocaleInfo(m_lcid, LOCALE_SNATIVELANGNAME, buf, WXSIZEOF(buf));
+                            wxString strLang = buf;
```

Comment:
We should add a helper function for this, i.e. something like
```cpp
wxString GetLocaleInfoString(LCTYPE lcType) const
{
    wxChar buf[256];
    buf[0] = wxT('\0');
    ::GetLocaleInfo(m_lcid, lcType, buf, WXSIZEOF(buf));
    return buf;
}
```
and use it both in the new (i.e. this) and existing code.

Reply:
Correction: we already have `wxUILocaleImplName::DoGetInfo()` so the helper here should be called `DoGetInfo()` too and should also be written in the same way, i.e. check for errors as well.

Reply:
I will modify the code accordingly.